internals-js/src/specs/sourceSpec.ts

```diff
@@ -172,17 +175,33 @@ export class SourceSpecDefinition extends FeatureDefinition {
             }
           });
         }
+        if (featureUrl && featureUrl.name === 'federation') {
+          federationVersion = featureUrl.version;
+        }
       });
 
+    if (result.sourceAPI || result.sourceType || result.sourceField) {
+      // Since this subgraph uses at least one of the @source{API,Type,Field}
+      // directives, it must also use v2.7 or later of federation.
+      if (!federationVersion || federationVersion.lt(this.minimumFederationVersion)) {
+        errors.push(ERRORS.SOURCE_FEDERATION_VERSION_REQUIRED.err(
+          `@source{API,Type,Field} directives require @link-importing federation ${
+            this.minimumFederationVersion
+          } or later`,
+        ));
```

Comment:
Maybe
```suggestion
          `@source{API,Type,Field} directives require federation version ${
            this.minimumFederationVersion
          } or later, found ${federationVersion}`,
        ));
``` ?

Reply:
See https://github.com/apollographql/federation/pull/2914/commits/264134ed930629e09d4ca84db4d1cd88caddea89 and https://github.com/apollographql/federation/pull/2914/commits/7ced7638606529424950b47d98fbb3338066ff50